vane-regions/src/main/java/org/oddlama/vane/regions/Regions.java

```diff
@@ -182,12 +185,20 @@ public class Regions extends Module<Regions> {
 	public RegionEconomyDelegate economy;
 
 	public Regions() {
-		menus = new RegionMenuGroup(this);
+		final var portals_plugin = get_module().getServer().getPluginManager().getPlugin("vane-portals");
```

Comment:
Thought that might be possible, didn't run into it so I didn't think too much of it. :smile:

Reply:
Not sure it's complete yet, the whole line must go away here. I think just below here it's still being used in the inconsistent state:

```java
vane_portals_available = portals_plugin != null;
menus = new RegionMenuGroup(this, vane_portals_available);
```

Reply:
It should be safe here since we're just checking its presence. It may not be loaded but it should be present since we softdepend.

Reply:
No, it only works because `p` is before `r` in the alphabet (so the jar file vane-portals is injected before vane-regions). If you were to rename the jar files such that the insertion order changes, this will break.

The constructor of a JavaPlugin is called immediately after it is injected into the classpath. The softdepend stuff is only relevant for the onEnable order, which is determined by bukkit. And before bukkit can know this order, it has to load each jar file.

You can only make this correct by delaying all plugin checks at least until on_enable, where all available jar files have been loaded.

Reply:
Yep, nevermind. I was going with https://www.spigotmc.org/wiki/plugin-yml/ which says "Your plugin will load after any plugins listed here" for softdepend.

An issue comes up though, just moving the assignment to `menus` into `onEnable` creates a `RegionMenuGroup` with all of the translated strings being set to `null`. Could that be coming from the lang reload prior to `on_enable` here: https://github.com/oddlama/vane/blob/81d26622f92fd53a798adba8594ccb157b5c3c0c/vane-core/src/main/java/org/oddlama/vane/core/module/Module.java#L360

I haven't looked into the lang file handling too much, but that's my best guess.

Reply:
Yes exactly. Vane's abstracted module logic has become quite sophisticated and does make some hidden assumptions...

Your best bet is definitely to remove the constructor parameter and leave everything as before. Then if you detect portals in on_enable, you can save the `RegionPortalIntegration` as a member variable and use `get_module().region_portal_integration != null` anywhere you need to actually do the check.